### PR TITLE
handles vdo volume reboot

### PR DESCRIPTION
**Pull request by ramkrsna**
- x-systemd.requires=vdo.service 
- handles reboot scenarios for vdo volumes

### CODE REVIEW

**Review by sac — APPROVED** [review]
Looks good to me. Have you completed this checklist:

* Tested for both VDO and non-vdo devices?
* Check fstab entries for VDO devices
* Check fstab entries for non-VDO devices
* Reboot checks:
   - Should start vdo service and mount vdo devices
   - Should mount non-vdo devices
   - Should not panic due to wrong fstab entries.

@ramkrsna if the above tests are done okay to merge.
@devyanikota request to review once.

**Review by devyanikota — COMMENTED** [review]
Hi, I have reviewed the patch. +1, looks good to me.
@ramkrsna have you tested the scenarios mentioned by sachi?

File: playbooks/mount.yml [hunk 1]
```diff
@@ -2,15 +2,22 @@
 - hosts: gluster_servers
   become: true
   gather_facts: no
+  vars:
+    opts="inode64,noatime,nodiratime"
```
Comment (sac) [inline playbooks/mount.yml:6]: This would cause an error. s/=/:/

File: playbooks/mount.yml [hunk 2]
```diff
@@ -2,15 +2,22 @@
...
 
   tasks:
   - name: Create the backend disks, skips if present
     file: path={{ item.path }} state=directory
     with_items: "{{ mntpath }}"
 
-  - name: Mount the volumes
+  - name: Set mount options for VDO
+    set_fact:
+      opts="{{opts}},"+"x-systemd.requires=vdo.service"
```
Comment (sac) [inline playbooks/mount.yml:15]: s/"+"//

**Review by sac — CHANGES_REQUESTED** [review]

**Comment by sac** [comment]
Closing this, will submit a new patch. The above errors cause boot hang.

**Pull request closed without merging**